### PR TITLE
[Impeller] Ensure known geometry has simple bounds computation.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Computing the bounds of an RRect can be really slow if you don't actually know that it is an RRect, and shows up in the traces for flutter gallery on wembly. This change should be semantically equivalent and is only an optimization to avoid recomputing the same value we already know.

### CODE REVIEW

File: impeller/aiks/canvas.cc [hunk 1]
```diff
@@ -273,10 +274,13 @@ void Canvas::DrawCircle(Point center, Scalar radius, const Paint& paint) {
...
   }
-  auto circle_path = PathBuilder{}
-                         .AddCircle(center, radius)
-                         .SetConvexity(Convexity::kConvex)
-                         .TakePath();
+  auto circle_path =
+      PathBuilder{}
+          .AddCircle(center, radius)
+          .SetConvexity(Convexity::kConvex)
+          .SetBounds(Rect::MakeLTRB(center.x - radius, center.y - radius,
```
Comment (zanderso) [inline impeller/aiks/canvas.cc:281]: Should this part also have a unit test?

File: impeller/aiks/canvas.cc [hunk 2]
```diff
@@ -252,6 +252,7 @@ void Canvas::DrawRRect(Rect rect, Scalar corner_radius, const Paint& paint) {
   auto path = PathBuilder{}
                   .SetConvexity(Convexity::kConvex)
                   .AddRoundedRect(rect, corner_radius)
+                  .SetBounds(rect)
```
Comment (gaaclarke) [inline impeller/aiks/canvas.cc:255]: So, the gist is that if you don't call SetBounds, it is getting calculated instead?  What about rounded rects that have extreme corner radii?  Does that become a shape whose bounds are different than the one specified in the draw call?

**Review by gaaclarke — COMMENTED** [review]
Did you measure it?

**Review by bdero — APPROVED** [review]
LGTM

File: impeller/aiks/canvas.cc (hunk 2 above)
Comment (jonahwilliams) [reply impeller/aiks/canvas.cc:255]: Yes, normally we use the Skia computed bounds - but the Aiks API doesn't accomidated that. What are you thinking of for extreme corner radii?

Comment (jonahwilliams) [reply impeller/aiks/canvas.cc:255]: Running some tests:

```
  for (auto i = - 100; i < 100; i += 10) {
    SkRRect rrect = SkRRect::MakeRectXY(SkRect::MakeLTRB(0, 0, 100, 100), i, i);
    SkPath path = SkPath().addRRect(rrect);
    auto bds = path.getBounds();
    FML_LOG(ERROR) << "Bounds for RRect with radius " << i << " -> " << bds.x() << "," << bds.y() << "m" << bds.width() << "," << bds.height();
  }
```  


Result:

```
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius -100 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius -90 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius -80 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius -70 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius -60 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius -50 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius -40 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius -30 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius -20 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius -10 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius 0 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius 10 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius 20 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius 30 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius 40 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius 50 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius 60 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius 70 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius 80 -> 0,0m100,100
[ERROR:flutter/impeller/display_list/dl_unittests.cc(1748)] Bounds for RRect with radius 90 -> 0,0m100,100
```

File: impeller/aiks/canvas.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/aiks/canvas.cc:281]: Done

File: impeller/aiks/canvas.cc (hunk 2 above)
Comment (gaaclarke) [reply impeller/aiks/canvas.cc:255]: I was wondering if when the corner radius was high would it start looking more like a circle that was inset in the rect.  LGTM.

Comment (flar) [reply impeller/aiks/canvas.cc:255]: Did you try this with reversed rects? RRects is one of the areas where Skia was doing "auto-sorting" of the corners of the rects...

**Pull request merged**